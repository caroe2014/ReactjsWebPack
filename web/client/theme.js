// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
