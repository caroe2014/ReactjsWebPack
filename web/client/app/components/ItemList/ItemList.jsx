// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component } from 'react';
import styled, { ThemeProvider } from 'styled-components';
import { Flex, Heading } from 'rebass';
import PropTypes from 'prop-types';
import theme from 'web/client/theme.js';
import Item from '../Item/Item';
import { Trans } from 'react-i18next';
import i18n from 'web/client/i18n';

const Container = styled(Flex)`
  flex-direction: column;
  margin: 120px auto 0px auto;
  height: fit-content;
  max-width: 75em;
  width: 100%;
  padding: 0px;
  &:after{
    content: '';
    margin: auto;
    width: 100%;
    max-width: 570px;
    }
  ${props => props.theme.mediaBreakpoints.mobile`
    justify-content: center;
    padding-left: 10px;
    padding-right: 10px;
  `}
  ${props => props.theme.mediaBreakpoints.tablet`
    justify-content: center;
  `}
  ${props => props.theme.mediaBreakpoints.desktop`
    justify-content: space-between;
    padding-left: 20px;
    padding-right: 20px;
  `}
`;

const ListContainer = styled(Flex)`
  flex-direction: column;
  flex-flow: row wrap;
  height: fit-content;
  max-width: 75em;
  width: 100%;
  padding: 0px;
  &:after{
    content: '';
    margin: auto;
    width: 100%;
    max-width: 570px;
    }
  ${props => props.theme.mediaBreakpoints.mobile`
    justify-content: center;
  `}
  ${props => props.theme.mediaBreakpoints.tablet`
    justify-content: center;
  `}
  ${props => props.theme.mediaBreakpoints.desktop`
    justify-content: space-between;
  `}
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const Toptext = styled(Heading)`
  width: 100%;
  color: ${props => props.theme.colors.primaryTextColor};
  font-size: 18px;
  margin-top: 20px;
  margin-bottom: 20px;
  font-weight: 700;
  ${props => props.theme.mediaBreakpoints.tablet`margin: 20px auto; max-width: 570px;`}
  ${props => props.theme.mediaBreakpoints.mobile`padding-left: 0px;`}
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

class ItemList extends Component {

  render () {

    const { keyProps, showAddButton, addItemToCart, selectItem, currencyDetails, validList, digitalMenuId, itemDisplayList } = this.props; // eslint-disable-line max-len

    return (
      <ThemeProvider theme={theme}>
        <Container className='item-list' p={[1, 2, 3, 4]} m={[0, 2]}>

          <Toptext className='select-item-header' role='heading' tabIndex={0} aria-level='2'>
            {!digitalMenuId && <Trans i18nKey='ITEM_LIST_PAGE_LABEL' />}</Toptext>
          <ListContainer className='item-listcontainer'
            tabIndex={0}
            aria-label={`${(keyProps && keyProps.length > 1)
              ? `${i18n.t('ITEM_SELECT')} ${keyProps.length} ${i18n.t('ITEMS')}` : ``}`}>
            {keyProps && validList && keyProps.map((item, index) => (
              <React.Fragment key={`item-${item.id.toString()}`} >
                <Item
                  className={`tile item-${index + 1}`}
                  keyProps={item}
                  showAddButton={showAddButton}
                  addItemToCart={addItemToCart}
                  selectItem={selectItem}
                  digitalMenuId={digitalMenuId}
                  currencyDetails={currencyDetails}
                  ariaLabel={`item${index + 1} ${item.displayText}`}
                  tabIndex={0}
                  itemDisplayList={itemDisplayList}
                />
              </React.Fragment>
            ))}
          </ListContainer>
        </Container>
      </ThemeProvider>
    );
  }
}

ItemList.propTypes = {
  keyProps: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      description: PropTypes.string,
      image: PropTypes.string,
      attributes: PropTypes.arrayOf(
        PropTypes.shape({
          id: PropTypes.string.isRequired,
          name: PropTypes.string.isRequired,
          value: PropTypes.string.isRequired
        }))
    }).isRequired
  ).isRequired,
  currencyDetails: PropTypes.object,
  itemDisplayList: PropTypes.array
};

export default ItemList;
