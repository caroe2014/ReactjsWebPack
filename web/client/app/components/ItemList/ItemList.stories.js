// Copyright © 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';

import { storiesOf } from '@storybook/react';
import ItemList from './ItemList';

storiesOf('ItemList Component', module)
  .add('With images', () => {
    const menuItems = [
      {
        id: 100,
        name: 'Mc Alister’s Club',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
        amount: '12.00',
        image: 'https://lh3.googleusercontent.com/-oCQXVe5cO98sceFwpt-9yYpkjTXX1Fb7ZfMFSCA1FFXRfJOTg2wCV1hPbyJO7HqexJYBN5frg5NhXVsgaFhG5I=s730-c-e365', // eslint-disable-line max-len
        attributes: [
          { id: 'gluten', name: 'Gluten Free', value: 'Gluten Free' },
          { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' }
        ]
      },
      {
        id: 101,
        name: 'King Club',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
        amount: '12.00',
        image: 'https://www.landolakes.com/RecipeManagementSystem/media/Recipe-Media-Files/Recipes/Retail/x17/20733-all-american-club-sandwich-600x600.jpg?ext=.jpg', // eslint-disable-line max-len
        attributes: [
          { id: 'gluten', name: 'Gluten Free', value: 'Gluten Free' },
          { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' },
          { id: 'calories', name: 'Calories', value: '120Cal' }
        ]
      },
      {
        id: 102,
        name: 'Big mc Burger',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
        amount: '12.00',
        image: 'https://imagesvc.timeincapp.com/v3/mm/image?url=http%3A%2F%2Fcdn-image.myrecipes.com%2Fsites%2Fdefault%2Ffiles%2Fstyles%2Fmedium_2x%2Fpublic%2F1426263343%2FBestEverJuicy-Burger_420x420.jpg%3Fitok%3DwNkrRxc4&w=700&q=85', // eslint-disable-line max-len
        attributes: [
          { id: 'gluten', name: 'Gluten Free', value: 'Gluten Free' },
          { id: 'calories', name: 'Calories', value: '120Cal' }
        ]
      },
      {
        id: 103,
        name: 'Non veg Club',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
        amount: '12.00',
        image: 'https://ls.imgix.net/recipes/thumbnails/BBQMatesAmerican-Club-Sandwich-with-Candied-Bacon.jpg?w=620&h=310&auto=compress,format&fit=crop', // eslint-disable-line max-len
        attributes: [
          { id: 'gluten', name: 'Gluten Free', value: 'Gluten Free' },
          { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' },
          { id: 'calories', name: 'Calories', value: '120Cal' }
        ]
      },
      {
        id: 104,
        name: 'Spicy south',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes. Sandwiches filled with nice cut veggies with slightly coated cheese.', // eslint-disable-line max-len
        amount: '12.00',
        image: 'https://bigoven-res.cloudinary.com/image/upload/t_recipe-480/classy-club-sandwiches-2.jpg',
        attributes: [
          { id: 'gluten', name: 'Gluten Free', value: 'Gluten Free' },
          { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' },
          { id: 'calories', name: 'Calories', value: '120Cal' }
        ]
      },
      {
        id: 105,
        name: 'Grilled chicken sandwich',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
        amount: '12.00',
        image: 'https://bigoven-res.cloudinary.com/image/upload/t_recipe-480/the-downtowner-sandwich-fbafcd-1ef83878f8097edf4f02867d.jpg', // eslint-disable-line max-len
        attributes: [
          { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' }
        ]
      },
      {
        id: 106,
        name: 'Cheese max pasta',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
        amount: '12.00',
        image: 'https://bigoven-res.cloudinary.com/image/upload/t_recipe-480/davinci-pasta-like-cheesecake-facto.jpg',
        attributes: [
          { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' },
          { id: 'calories', name: 'Calories', value: '120Cal' }
        ]
      }
    ];

    return (
      <ItemList keyProps={menuItems} showAddButton/>
    );
  }
  );

storiesOf('ItemList Component', module)
  .add('Without images', () => {
    const menuItems = [
      {
        id: 100,
        name: 'Mc Alister’s Club',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
        amount: '12.00',
        attributes: [
          { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' },
          { id: 'calories', name: 'Calories', value: '120Cal' }
        ]
      },
      {
        id: 101,
        name: 'Mc Alister’s Club',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
        amount: '12.00',
        attributes: [
          { id: 'gluten', name: 'Gluten Free', value: 'Gluten Free' },
          { id: 'calories', name: 'Calories', value: '120Cal' }
        ]
      },
      {
        id: 102,
        name: 'King Club',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
        amount: '12.00',
        attributes: [
          { id: 'gluten', name: 'Gluten Free', value: 'Gluten Free' },
          { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' },
          { id: 'calories', name: 'Calories', value: '120Cal' }
        ]
      },
      {
        id: 103,
        name: 'Non veg Club',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
        amount: '12.00',
        attributes: [
          { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' }
        ]
      },
      {
        id: 104,
        name: 'Spicy south',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes. Sandwiches filled with nice cut veggies with slightly coated cheese.', // eslint-disable-line max-len
        amount: '12.00',
        attributes: [
          { id: 'gluten', name: 'Gluten Free', value: 'Gluten Free' },
          { id: 'healthy', name: 'Healthy Choice', value: 'Healthy' },
          { id: 'calories', name: 'Calories', value: '120Cal' }
        ]
      },
      {
        id: 105,
        name: 'Grilled chicken sandwich',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
        amount: '12.00',
        attributes: [
          { id: 'gluten', name: 'Gluten Free', value: 'Gluten Free' },
          { id: 'calories', name: 'Calories', value: '120Cal' }
        ]
      },
      {
        id: 106,
        name: 'Cheese max pasta',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
        amount: '12.00',
        attributes: [
          { id: 'type', value: 'Veg' },
          { id: 'gluten', name: 'Gluten Free', value: 'Gluten Free' },
          { id: 'calories', name: 'Calories', value: '120Cal' }
        ]
      },
      {
        id: 107,
        name: 'Chicken tofu',
        description: 'Organic and natural sandwiches with delicious sauces made in all sizes',
        amount: '12.00',
        attributes: [
          { id: 'calories', name: 'Calories', value: '120Cal' }
        ]
      }
    ];
    return (
      <ItemList keyProps={menuItems} showAddButton/>
    );
  }
  );
