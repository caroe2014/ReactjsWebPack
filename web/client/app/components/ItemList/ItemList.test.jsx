// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme.js';
import ItemList from './ItemList';
import { t } from 'i18next';

configure({ adapter: new Adapter() });

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

describe('ItemList Component', () => {

  let wrapper;
  let mockProps = [
    {
      id: '100',
      name: 'This is a mock name',
      description: 'Mock description text',
      amount: 'Mock amount',
      image: 'image.png',
      attributes: [
        { id: 'mockId1', name: 'mockName', value: 'mockValue1' },
        { id: 'mockId2', name: 'mockName', value: 'mockValue2' },
        { id: 'mockId3', name: 'mockName', value: 'mockValue3' }
      ]
    },
    {
      id: '101',
      name: 'This is a mock name',
      description: 'Mock description text',
      amount: 'Mock amount',
      image: 'image.png',
      attributes: [
        { id: 'mockId1', name: 'mockName', value: 'mockValue1' },
        { id: 'mockId2', name: 'mockName', value: 'mockValue2' },
        { id: 'mockId3', name: 'mockName', value: 'mockValue3' }
      ]
    },
    {
      id: '102',
      name: 'This is a mock name',
      description: 'Mock description text',
      amount: 'Mock amount',
      image: 'image.png',
      attributes: [
        { id: 'mockId1', name: 'mockName', value: 'mockValue1' },
        { id: 'mockId2', name: 'mockName', value: 'mockValue2' }
      ]
    },
    {
      id: '103',
      name: 'This is a mock name',
      description: 'Mock description text',
      amount: 'Mock amount',
      image: 'image.png',
      attributes: [
        { id: 'mockId1', name: 'mockName', value: 'mockValue1' },
        { id: 'mockId2', name: 'mockName', value: 'mockValue2' }
      ]
    }
  ];

  beforeEach(() => {
    wrapper = mount(<ThemeProvider theme={theme}><ItemList keyProps={mockProps} /></ThemeProvider>);
  });

  it('does a snapshot test', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(wrapper.exists()).toBe(true);
  });

  describe('when Item props are received', () => {
    it('should render the same number of tiles', () => {
      const ListWrapper = shallow(<ItemList keyProps={mockProps} validList/>);
      expect(ListWrapper.find('.tile').length).toBe(mockProps.length);
    });
  });

  describe('when Item props are not received', () => {
    beforeEach(() => {
      mockProps = [];
    });
    it('should not render ItemComponent', () => {
      const itemWrapper = shallow(<ItemList keyProps={mockProps} />);
      expect(itemWrapper.find('.tile').length).toBe(0);
    });
  });

});
