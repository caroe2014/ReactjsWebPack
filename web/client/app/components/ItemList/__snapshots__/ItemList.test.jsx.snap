// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList Component does a snapshot test 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <ItemList
    keyProps={
      Array [
        Object {
          "amount": "Mock amount",
          "attributes": Array [
            Object {
              "id": "mockId1",
              "name": "mockName",
              "value": "mockValue1",
            },
            Object {
              "id": "mockId2",
              "name": "mockName",
              "value": "mockValue2",
            },
            Object {
              "id": "mockId3",
              "name": "mockName",
              "value": "mockValue3",
            },
          ],
          "description": "Mock description text",
          "id": "100",
          "image": "image.png",
          "name": "This is a mock name",
        },
        Object {
          "amount": "Mock amount",
          "attributes": Array [
            Object {
              "id": "mockId1",
              "name": "mockName",
              "value": "mockValue1",
            },
            Object {
              "id": "mockId2",
              "name": "mockName",
              "value": "mockValue2",
            },
            Object {
              "id": "mockId3",
              "name": "mockName",
              "value": "mockValue3",
            },
          ],
          "description": "Mock description text",
          "id": "101",
          "image": "image.png",
          "name": "This is a mock name",
        },
        Object {
          "amount": "Mock amount",
          "attributes": Array [
            Object {
              "id": "mockId1",
              "name": "mockName",
              "value": "mockValue1",
            },
            Object {
              "id": "mockId2",
              "name": "mockName",
              "value": "mockValue2",
            },
          ],
          "description": "Mock description text",
          "id": "102",
          "image": "image.png",
          "name": "This is a mock name",
        },
        Object {
          "amount": "Mock amount",
          "attributes": Array [
            Object {
              "id": "mockId1",
              "name": "mockName",
              "value": "mockValue1",
            },
            Object {
              "id": "mockId2",
              "name": "mockName",
              "value": "mockValue2",
            },
          ],
          "description": "Mock description text",
          "id": "103",
          "image": "image.png",
          "name": "This is a mock name",
        },
      ]
    }
  >
    <ThemeProvider
      theme={
        Object {
          "colors": Object {
            "addOnTab": "#347CFF",
            "addToCart": "#347CFF",
            "addonBg": "#EEEEEE",
            "backgroundGrey": "#E6E6E6",
            "black": "#000000",
            "close": "red",
            "contextBackground": "#f0f0f3",
            "darkGrey": "#464646",
            "disableBackground": "#E4E4E4",
            "disableButton": "#D1D1D1",
            "error": "#cc0000",
            "footer": "#48a942",
            "glutenFree": "#8d7356",
            "healthy": "#359919",
            "hint": "#a6a6a6",
            "inputBottomBorder": "lightgray",
            "itemBorder": "lightgrey",
            "itemTile": "#3C474C0",
            "light": "white",
            "loader": "#ababab",
            "loginBackground": "#303030",
            "loginBoxColor": "#434149",
            "loginField": "#AAA7AC",
            "loginFormBackground": "#484848",
            "navigationMenu": "#48a942",
            "navigationMenuHover": "cornflowerblue",
            "new": "#ff6000",
            "open": "#48a942",
            "paymentSuccessTitle": "#505050",
            "placeHolder": "#909090",
            "primary": "#347CFF",
            "primaryButton": "#347CFF",
            "primaryText": "#383838",
            "secondaryText": "#a6a6a6",
            "spicy": "#ed1c24",
            "text": "black",
            "textGrey": "#505050",
            "tile": "#505050",
            "validationError": "#cc0000",
            "vegan": "#0d4a22",
            "vegetarian": "#0d4a22",
          },
          "fontFamily": "Roboto",
          "fontSize": Object {
            "lg": "1.8rem",
            "md": "1.3rem",
            "nm": "1rem",
            "sm": "0.75rem",
            "xl": "2.4rem",
          },
          "fontSizeFromProps": [Function],
          "mediaBreakpoints": Object {
            "desktop": [Function],
            "mobile": [Function],
            "tablet": [Function],
          },
          "nav": Object {
            "height": "70px",
          },
          "receiptFontFamily": "Roboto",
          "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
          "spacing": Object {
            "lg": "4em",
            "md": "2em",
            "nm": "1em",
            "none": "0em",
            "sm": "0.5em",
            "xl": "8em",
          },
        }
      }
    >
      <Styled(Flex)
        className="item-list"
        m={
          Array [
            0,
            2,
          ]
        }
        p={
          Array [
            1,
            2,
            3,
            4,
          ]
        }
      >
        <Flex
          className="item-list sc-dqBHgY iyTuOp"
          m={
            Array [
              0,
              2,
            ]
          }
          p={
            Array [
              1,
              2,
              3,
              4,
            ]
          }
        >
          <Box
            className="item-list sc-dqBHgY iyTuOp sc-bwzfXH hKiLMS"
            m={
              Array [
                0,
                2,
              ]
            }
            p={
              Array [
                1,
                2,
                3,
                4,
              ]
            }
          >
            <div
              className="item-list sc-dqBHgY iyTuOp sc-bwzfXH hKiLMS sc-bdVaJa dQThJL"
            >
              <Styled(Heading)
                aria-level="2"
                className="select-item-header"
                role="heading"
                tabIndex={0}
              >
                <Heading
                  aria-level="2"
                  className="select-item-header sc-dfVpRl edrqND"
                  fontSize={5}
                  fontWeight="bold"
                  is="h2"
                  lineHeight={1.25}
                  m={0}
                  role="heading"
                  tabIndex={0}
                >
                  <ForwardRef
                    aria-level="2"
                    className="select-item-header sc-dfVpRl edrqND sc-VigVT fXkMTo"
                    fontSize={5}
                    fontWeight="bold"
                    is="h2"
                    lineHeight={1.25}
                    m={0}
                    role="heading"
                    tabIndex={0}
                  >
                    <h2
                      aria-level="2"
                      className="select-item-header sc-dfVpRl edrqND sc-VigVT fXkMTo"
                      role="heading"
                      tabIndex={0}
                    >
                      <div
                        i18nKey="ITEM_LIST_PAGE_LABEL"
                      />
                    </h2>
                  </ForwardRef>
                </Heading>
              </Styled(Heading)>
              <Styled(Flex)
                aria-label="ITEM_SELECT 4 ITEMS"
                className="item-listcontainer"
                tabIndex={0}
              >
                <Flex
                  aria-label="ITEM_SELECT 4 ITEMS"
                  className="item-listcontainer sc-gxMtzJ ftlvFG"
                  tabIndex={0}
                >
                  <Box
                    aria-label="ITEM_SELECT 4 ITEMS"
                    className="item-listcontainer sc-gxMtzJ ftlvFG sc-bwzfXH hKiLMS"
                    tabIndex={0}
                  >
                    <div
                      aria-label="ITEM_SELECT 4 ITEMS"
                      className="item-listcontainer sc-gxMtzJ ftlvFG sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
                      tabIndex={0}
                    />
                  </Box>
                </Flex>
              </Styled(Flex)>
            </div>
          </Box>
        </Flex>
      </Styled(Flex)>
    </ThemeProvider>
  </ItemList>
</ThemeProvider>
`;
